src/ui/Analytics/Analytics.ts

```diff
@@ -463,6 +469,14 @@ export class Analytics extends Component {
     );
   }
 
+  private handleCoveoUAEventReady(data: ICoveoUAEventArgs) {
+    if (this.options.populateDataLayer) {
+      let dataLayerName = this.options.dataLayerName;
+      (<any>window)[dataLayerName] = (<any>window)[dataLayerName] || [];
```

Comment:
You can augment the global window interface to declare that it contains a property for `dataLayerName` instead of having to cast the global window object to any.

https://stackoverflow.com/a/12709880



Is that global variable declared on "window" really changing and different for each website using google tag manager ? 

Could we set a good default value for the option ?

We should also handle that the option might be misconfigured, and that `window[dataLayerName] == null`, or that `window[dataLayerName]` is somehow something totally different that google tag manager (and that the `push` function would not be available).

Reply:
I wouldn't augment the window interface in this case, because we can't be a 100% sure the property is going to be named `dataLayer`.

It is possible for users to rename the dataLayer in each website, though I doubt this is common practice. The current default (`dataLayer`) is also the name that is output by default when you generate GTM code snippets, so I think that's a good default value.

I'm adding some logic to ensure that the dataLayer is an object, and that it contains at least an item with the `gtm.uniqueEventId` property before pushing anything.

I'm also adding logic to assert that a GTM code snippet is present in the page before even registering the event handler.

Reply:
(about renaming the dataLayer: https://developers.google.com/gtagjs/devguide/rename)